src/main.rs

```diff
@@ -108,8 +96,10 @@ async fn main() -> std::io::Result<()> {
         }
     };
 
+    let db_args = args.clone();
+    let db = get_database(db_args.into());
     let app_state = AppState {
-        pastas: Arc::new(Mutex::new(read_all(&args))),
```

Comment:
Your arcing (wrapping it in an Arc) the database here but its already in a box. You can instead Arc it in get_database. (return Arc<dyn traits>).